Scripts/Tracking/Velocity/XRNodeVelocityEstimator.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace VRTK.Core.Tracking.Velocity
+{
+    using System.Collections.Generic;
+    using UnityEngine;
+    using UnityEngine.XR;
+
+    /// <summary>
+    /// Retrieves the velocity estimations for the specified <see cref="XRNode"/> device.
+    /// </summary>
+    public class XRNodeVelocityEstimator : VelocityTracker
+    {
+        /// <summary>
+        /// The device to query velocity estimations for.
+        /// </summary>
+        [Tooltip("The device to query velocity estimations for.")]
+        public XRNode device = XRNode.LeftHand;
+
+        /// <inheritdoc />
+        public override bool IsActive()
+        {
+            return isActiveAndEnabled;
+        }
+
+        /// <inheritdoc />
+        protected override Vector3 DoGetVelocity()
+        {
+            Vector3 result = Vector3.zero;
+            GetNodeState().TryGetVelocity(out result);
+            return result;
+        }
+
+        /// <inheritdoc />
+        protected override Vector3 DoGetAngularVelocity()
+        {
+            Vector3 result = Vector3.zero;
+            GetNodeState().TryGetAngularVelocity(out result);
+            return result;
+        }
+
+        /// <summary>
+        /// Gets the state of the node associated with the specified <see cref="XRNode"/> device.
+        /// </summary>
+        /// <returns>The associated node state.</returns>
+        protected virtual XRNodeState GetNodeState()
+        {
+            List<XRNodeState> nodesStates = new List<XRNodeState>();
+            InputTracking.GetNodeStates(nodesStates);
```

Comment:
```suggestion
            return InputTracking.GetNodeStates(nodesStates).FirstOrDefault(state => state.nodeType == node);
```
Then remove the rest of this method.

Reply:
That doesn't work as `GetNodeStates` returns void, it's basically piggy backing on nodeStates being a reference parameter. it's so terrabad I can't begin to comprehend the reason why it works that way!

anyway this works

```
nodesStates.Clear();
InputTracking.GetNodeStates(nodesStates);
return nodesStates.FirstOrDefault(state => state.nodeType == node);
```

(i'm just clearing the nodeStates list first as it's now an instance variable and I dunno how clean it is so a quick clear shouldn't hurt)

Reply:
According to the [docs](https://docs.unity3d.com/ScriptReference/XR.InputTracking.GetNodeStates.html) Unity does the same so we shouldn't call `Clear` to prevent that unnecessary additional call.